### ISSUE TITLE
Entering "~" as search string breaks lunr.

### ISSUE THREAD

**Issue by karland** [olivernn/lunr.js#274]
Searching only for `"~"` breaks lunr.js with the following error message:

"QueryParseError: expected either a field or a term, found EDIT_DISTANCE with value '1'"

I know, that `"~"` is used to indicate fuzzy search, but a user does not know this, if he enters `"~"` into a search input field.

**Comment by olivernn** [comment]
I'm not sure what more Lunr can do here. The string "~" is an invalid query string. Lunr reports this by throwing a specific exception, `lunr.QueryParseError`, provides an explanation and indicates where the offending part of the query is.

It is up to the application to use this information to present the user with an appropriate response to the invalid query. If there is information that Lunr _isn't_ providing that would make this easier then let me know, otherwise I think this is something that must be handled in your application.

**Comment by karland** [comment]
@olivernn Thank you for your fast response. I do not take this for granted.

I am also not sure what the best strategy is. Here are my thoughts: 

+ If Lunr is throwing exceptions which need to be handled by the app that is using Lunr, it should be clearly documented and in the example. Neither in the docs nor in the example nor in any tutorials have I seen that the integration of Lunr requires more than generating the index and searching for it. But maybe I overlooked something. 

+ For those who are unaware that the app needs to deal with invalid query strings, Lunr IMHO should rather issue a warning/error on the console.log. If an invalid character is input into a search field the default behaviour should rather be to ignore the invalid character by default, instead of break the whole app which is the worst case to me.

+ with respect to the `"~"` I believe the error is only thrown if `"~"` is the first character. This should be easy to deal with. Are there more characters / cases like this that need to be dealt with by the app?

What do you think?

**Comment by olivernn** [comment]
To be fair the documentation does mention that the [`lunr.Index#search`](https://lunrjs.com/docs/lunr.Index.html#search) method throws a `lunr.QueryParseError` if the query string cannot be parsed. In addition the [demo code](https://github.com/olivernn/moonwalkers/blob/master/src/main.js#L93-L102) _does_ handle the error, which is why it is displayed when performing the search you mention.

On your second point, I disagree. Lunr should report that something went wrong. No application code can respond to a log message. However, if you don't catch the exception the browser will print the exception in the log anyway. In your application you can of course silently ignore invalid queries, that is your choice, but Lunr can't make that choice for every application, so when there is an error, it must raise that error.

Lunr will throw a `lunr.QueryParseError` whenever there is an error parsing the query, having a "~" as the first character is just one of several examples of an invalid query. There is nothing special about this particular case.

Ultimately this behaviour is not going to change, an invalid query will continue to throw a `lunr.QueryParseError` which the application can choose to handle however they wish. I would recommend handling it by catching the exception and displaying an appropriate error message to the user, but that is completely up to you.

**Comment by karland** [comment]
Thanks again for the quick response. Indeed, I have not seen the two source you provided, which is certainly my fault, but on the other hand maybe they are not so obvious to spot. Maybe a short section in Guides with a header such as "Integration into an app" would have been helpful. 

> On your second point, I disagree. Lunr should report that something went wrong. No application code can respond to a log message. However, if you don't catch the exception the browser will print the exception in the log anyway. 

I actually agree with what you are saying. The only point I had was that I did not catch the exception because I overlooked / did not know that this was necessary and my whole React app got killed. Now that I know, it is easy to catch the exception.

I thought I mention this, but I see your points. But anyway, thanks for your support. I very much appreciate it. I am happy to close the issue.
